apps/src/blockly/addons/cdoFieldVariable.ts

```diff
@@ -112,14 +114,25 @@ export default class CdoFieldVariable extends GoogleBlockly.FieldVariable {
     options.pop();
     options.pop();
 
+    const filteredOptions = options.filter(option => {
+      const workspace = this.getSourceBlock()?.workspace;
+      if (!workspace) {
+        return true;
+      }
+
+      const nonParamVarIds = getNonFunctionVariableIds(workspace);
```

Comment:
Would it be worth moving some of this logic outside of the filter? For example, if there's no workspace, we don't need to filter the options at all. Can we also move `getNonFunctionVariableIds` outside of the filter so it doesn't need to be called for every option?

Reply:
Good call on `getNonFunctionVariableIds`. I think the rest of it needs to stay because we're processing the individual toolbox item.